src/components/DataGridWithInfiniteScroll/DataGridWithInfiniteScroll.tsx

```diff
@@ -0,0 +1,862 @@
+     * @param {prepareRow} is a function that must be called on any row to be displayed. It is responsible for preparing a row for rendering.
+     * @param {getToggleHideAllColumnsProps} If datagrid have hide show column functionality
+     * @param {useSortBy} to add sorting functionality on datagrid
+     * @param {useResizeColumns} this is default functionality of react table to provide column resize
+     * @param {useTable } hook takes options and plugins to build a table instance. The basic options are columns and row data.
+     * @param {useBlockLayout }is a react-table plugin hook that adds support for headers and cells to be rendered as inline-block divs (or other non-table elements) with explicit width.
+     * @param {getSortByToggleProps } getSortByToggleProps() is a function that resolves any props needed while toggling the sort direction.
+     */
+    const {
+        getTableProps,
+        getTableBodyProps,
+        headerGroups,
+        rows,
+        allColumns,
+        prepareRow,
+        getToggleHideAllColumnsProps,
+    } = useTable(
+        {
+            columns,
+            data,
+        },
+        useSortBy,
+        useBlockLayout,
+        useResizeColumns
+    );
+
+    const [allValues, setIsChecked] = useState<any>([]);
+    useEffect(() => {
+        if (rows.length !== 0) {
+            setIsChecked(rows);
+        }
+    }, [rows]);
+
+    /**
+     * returnPositionColumnSelectionDropdown function
+     * This function returns the positions of the column selection dropdown as the table size varies (columns hidden/shown)
+     */
+    const returnPositionColumnSelectionDropdown = () => {
+        if (
+            refParent.current !== null &&
+            refParent.current !== undefined &&
+            refChild.current !== undefined &&
+            refChild.current !== null
+        ) {
+            const HideShowColumnsMenuTop =
+                refParent.current.getClientRects()[0].top +
+                Constants.COLUMN_SELECTION_MENU_TOP;
+            const HideShowColumnsMenuLeft =
+                refParent.current.getClientRects()[0].width -
+                Constants.COLUMN_SELECTION_MENU_LEFT;
+            const transformVal =
+                "translateX(" +
+                HideShowColumnsMenuLeft +
+                "px) " +
+                "translateY(" +
+                HideShowColumnsMenuTop +
+                "px)";
+            settransformVal(transformVal);
+        }
+    };
+
+    /**
+     * getColumnSelectionList function
+     * This function is called on setting icon, to open the column selection dropdown
+     */
+    const getColumnSelectionList = () => {
+        returnPositionColumnSelectionDropdown();
+        setOpen(!isOpen);
+    };
+
+    /**
+     * handleDetailsPanel function
+     * This function returns the details for detail panel content
+     */
+    const handleDetailsPanel = (row: Row, i: number) => {
```

Comment:
```suggestion
    const handleDetailsPanel = (row: Row, index: number) => {
```
avoid using short forms like i , j or e